invoice and payments model done

connorfrendt opened the pull request.

Commit:
invoice and payments model done

Commit:
invoice and payments model done

blueridger commented on hifireg/registration/models/registration.py:
```diff
@@ -17,15 +36,14 @@ class Registration(models.Model):
     def is_submitted(self):
         return self.order_set.filter(session=None).exists()
 
-    @property
```
These property tags needed to be added back

tiftaylor committed:
Fixed product_card columns

tiftaylor committed:
Disable click while product card is in progress

blueridger commented on hifireg/registration/models/registration.py:
```diff
@@ -4,6 +4,25 @@
 from django.db.models import F
 from .comp_code import CompCode
 
+class RegistrationQuerySet(models.QuerySet):
```
Hrm, this isn't supposed to be in this PR, right?

bryant-ferguson committed:
Fix price display on order subtotal page

Also added the "dollars" filter that converts a price from our models (in cents) to dollars with a dollar sign and commas.
More generically, we could create a "format" filter that will accept any f-string

bryant-ferguson committed:
Remove the rest of AP calc page

bryant-ferguson committed:
Fix $0 ap_eligible bug

If $0 of your order is eligible for AP, this should not break the AP calculator.

bryant-ferguson committed:
Prevent access to subtotal page is subtotal is $0

Also added "DispatchMixin" mixin that allows injection of dispatch mixin code anywhere in MRO.
This is an experiment to see if perhaps it makes more sense to keep certain forceful browsing code
inside the view logic.

bryant-ferguson committed:
Prevent forceful acquisition of accessible pricing

By moving this logic into dispatch_mixin, we check for "is_accessible_pricing"
before any http method routing can take place either GET or POST. This
makes it impossible to forcefully POST if you don't already have accessible pricing.

bryant-ferguson committed:
Fix "make test"

bryant-ferguson committed:
clean up some things

bryant-ferguson committed:
Dispatch mixin chaining via decorator

Boo-ya!

bryant-ferguson committed:
Add NonZeroOrderRequiredMixin

tiftaylor committed:
Corrected Indentation

tiftaylor committed:
Div Style Consistency for All Registration Pages

Encompaess Buttons & Title on every Reg page (like originally on the Policy page)

Commit:
RegistrationQuerySet with_outstanding_balances

tiftaylor committed:
Replace FBV with CBV / Removed @decorators to Mixins

bryant-ferguson committed:
Add links for some navbar stuff

It's nice to have these to move around quickly when testing stuff out.

bryant-ferguson committed:
Formatting and comments

bryant-ferguson committed:
Add Mixin for policy agreement

Commit:
removing invoice from __init__

Commit:
hello world page working

tiftaylor committed:
Updated Order Model w/Donation Field and RegisterDonateView

bryant-ferguson committed:
Use RegistrationRequiredMixin instead of PolicyRequiredMixin for Policy page

Otherwise you will get into a redirect loop since this is the page that allows you to agree to policy.

bryant-ferguson committed:
Use free context for Index view

dependabot[bot] committed:
Bump django from 3.0.5 to 3.0.7

Bumps [django](https://github.com/django/django) from 3.0.5 to 3.0.7.
- [Release notes](https://github.com/django/django/releases)
- [Commits](https://github.com/django/django/compare/3.0.5...3.0.7)

Signed-off-by: dependabot[bot] <support@github.com>

bryant-ferguson committed:
Attach listener via javascript instead of HTML to simplify donation page

bryant-ferguson committed:
Update dependencies

tiftaylor committed:
Task 1 of 5 | Payment + Stripe | Landing Page

tiftaylor committed:
Task 2 of 5 | Payment & Stripe | Checkout Sessions

Commit:
importing invoice

Commit:
fixing merge conflicts
